NSS-Therapeutic-Raccoons/WorkforceManagement
# Rb employee edit

## Link to Ticket
Closes # [5](https://github.com/NSS-Therapeutic-Raccoons/WorkforceManagement/issues/5)

## Description of Proposed Changes
- Created mvc GET/POST for Employees/Edit/
- Created integration test for Employees/Edit/
- Built out Bangazon.sql to include more that 1 Training Program, and added several more Computers, so that testing can be verifed more readily.


## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout rb-employee-edit
```

run all tests and verify in browser


``ADD appsettings.json``
``ADD Config.cs``
``Put connection string in all proper places``


Run the server and play around with the edit feature to see if you can break it.


## Impacted Areas in Application

List general components of the application that this PR will affect:

* `EmployeeController.cs` MVC
* `EmployeeTest.cs` IntegrationTest
* `TrainingProgram.cs` Model (added Name)
* `Computer.cs` Model (added Make, Manufacturer)
* `EmployeeEditViewModel.cs` (seperated AddEdit into two, built massive ViewModel for Edit)
* `EmployeeAddEditViewModel.cs` (renamed to EmployeeAddViewModel per seperation)
* `Bangazon.sql` (added more data for better, more thorough testing)
* `Views/Employee/Edit.cshtml` (to properly display the new content per ticket request).

## Mentions @username

@khardt
@figamus
@jeremiah3643
@ricky-bruner
@askingalot

## Definition of Done

1. The Files must be fully documented. This includes the following:
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement of Ticket.
Every line of code has been peer reviewed.
1. Every test must run properly.
